src/webgl-surface/drawing/shape/ribbon-shape.ts

```diff
@@ -0,0 +1,113 @@
+import { ICurvedLineOptions } from '../../primitives/curved-line';
+import { IPoint } from '../../primitives/point';
+import { bezier2 } from '../../util/interpolation';
+import { ReferenceColor } from '../reference/reference-color';
+import { CurvedLineShape } from '../shape/curved-line-shape';
+
+export interface IRibbonOptions extends ICurvedLineOptions{
+  cachedQuadSegments?: boolean;
+  depth?: number;
+  startColor?: ReferenceColor;
+  endColor?: ReferenceColor;
+  start: IPoint;
+  start2: IPoint;
+  end: IPoint;
+  end2: IPoint;
+  resolution?: number;
+  center1: IPoint;
+  center2: IPoint;
+  controlPoints: IPoint[];
+}
+
+export class RibbonShape<T> extends CurvedLineShape<T> {
```

Comment:
Add a class comment to this to document the shape so people using the class can identify what this class is used for.